packages/flutter_tools/lib/executable.dart

```diff
@@ -42,16 +43,54 @@ import 'src/commands/test.dart';
 import 'src/commands/trace.dart';
 import 'src/commands/update_packages.dart';
 import 'src/commands/upgrade.dart';
+import 'src/crash_reporting.dart';
 import 'src/devfs.dart';
 import 'src/device.dart';
 import 'src/doctor.dart';
 import 'src/globals.dart';
 import 'src/ios/mac.dart';
 import 'src/ios/simulators.dart';
 import 'src/run_hot.dart';
+import 'src/runner/flutter_command.dart';
 import 'src/runner/flutter_command_runner.dart';
 import 'src/usage.dart';
 
+/// Command-line options shared across all commands.
+class TopLevelOptions {
+  final bool verboseHelp;
+
+  TopLevelOptions({@required this.verboseHelp});
```

Comment:
Actually, if you switch to the `run()` design suggested below, you don't need `TopLevelOptions` anymore.